src/main/java/com/example/demo/services/BookingService.java

```diff
@@ -0,0 +1,93 @@
+package com.example.demo.services;
+
+import com.example.demo.controllers.BookingController;
+import com.example.demo.entities.BookingEntity;
+import com.example.demo.entities.CarEntity;
+import com.example.demo.entities.CustomerEntity;
+import com.example.demo.entities.LocationEntity;
+import com.example.demo.error.IdNotFoundException;
+import com.example.demo.error.IllegalBookingException;
+import com.example.demo.repositories.BookingRepository;
+import com.example.demo.repositories.CarRepository;
+import com.example.demo.repositories.CustomerRepository;
+import com.example.demo.repositories.LocationRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.hateoas.mvc.ControllerLinkBuilder;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class BookingService implements IBookingService {
+    @Autowired BookingRepository bookingRepository;
+    @Autowired CarRepository carRepository;
+    @Autowired CustomerRepository customerRepository;
+    @Autowired LocationRepository locationRepository;
+
+    @Override
+    public ResponseEntity createBooking(BookingEntity booking) {
```

Comment:
Ok for this exercise, but consider if you wanted to reuse this method elsewhere in your code, the return type of ReponseEntity limits how useful it is. More common is to return the created entity, seeing as the alternative is an exception, and then other code which may need to automatically create a booking could call this, get the booking and use it immediately, instead of having to infer it from the ResponseEntity